File: vibeShare/src/main/java/com/vibeshare/Service/BookmarkService.java
package com.vibeshare.Service;

import com.vibeshare.DTO.Response.BookmarkResponse;

import java.util.List;

public interface BookmarkService {

    BookmarkResponse addBookmark(Long postId);

    void removeBookmark(Long postId);

    List<BookmarkResponse> getMyBookmarks();

    boolean isBookmarked(Long postId);
}

File: frontend/src/api/bookmark.service.ts
import { apiFetch } from "./client";
import type { BookmarkResponse } from "@/types/BookmarkResponse";

export const bookmarkService = {
  addBookmark: (postId: number) => {
    return apiFetch<BookmarkResponse>(`/api/bookmarks/${postId}`, {
      method: "POST",
    });
  },

  removeBookmark: (postId: number) => {
    return apiFetch<void>(`/api/bookmarks/${postId}`, {
      method: "DELETE",
    });
  },

  isBookmarked: (postId: number) => {
    return apiFetch<boolean>(`/api/bookmarks/${postId}/check`);
  },

  getMyBookmarks: () => {
    return apiFetch<BookmarkResponse[]>(`/api/bookmarks`);
  },
};
